examples/using-clickhouse/store/store_test.go

```diff
@@ -0,0 +1,212 @@
+package store
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	"github.com/ClickHouse/clickhouse-go/v2/lib/driver"
+	"github.com/google/uuid"
+	"github.com/stretchr/testify/assert"
+
+	clickhousemock "github.com/srikanthccv/ClickHouse-go-mock"
+
+	"gofr.dev/examples/using-clickhouse/models"
+	"gofr.dev/pkg/datastore"
+	"gofr.dev/pkg/errors"
+	"gofr.dev/pkg/gofr"
+)
+
+func TestUser_Create(t *testing.T) {
+	ctx := gofr.NewContext(nil, nil, gofr.New())
+	ctx.Context = context.Background()
+
+	uid := uuid.MustParse("37387615-aead-4b28-9adc-78c1eb714ca2")
+
+	store := New()
+
+	tests := []struct {
+		desc    string
+		id      string
+		user    models.User
+		resp    models.User
+		mockErr error
+		err     error
+	}{
+		{"Valid case", "1", models.User{Name: "stella", Age: "21"}, models.User{ID: uid, Name: "stella", Age: "21"}, nil, nil},
+		{"db error", "2", models.User{Name: "sony", Age: "21"}, models.User{ID: uid, Name: "sony", Age: "21"},
+			errors.DB{Err: errors.Error("database error")}, errors.DB{Err: errors.Error("database error")}},
+		{desc: "scan error", id: "5", user: models.User{Name: "stella", Age: "21"}, err: errors.DB{Err: errors.Error("scan error")}},
+	}
+
+	for i, tc := range tests {
+		ctx.DataStore = datastore.DataStore{ClickHouse: datastore.ClickHouseDB{Conn: mockClickhouse{id: tc.id}}}
+
+		_, err := store.Create(ctx, tc.user)
+
+		assert.Equal(t, tc.err, err, "TEST[%d], failed.\n%s", i, tc.desc)
+	}
+}
+
+func TestUser_Get(t *testing.T) {
+	ctx := gofr.NewContext(nil, nil, gofr.New())
+	ctx.Context = context.Background()
+
+	uid := uuid.MustParse("37387615-aead-4b28-9adc-78c1eb714ca2")
+	uid1 := uuid.MustParse("37387615-aead-4b28-9adc-78c1eb714ca4")
+
+	store := New()
+
+	tests := []struct {
+		desc string
+		id   string
+		resp []models.User
+		err  error
+	}{
+		{desc: "Valid case", id: "1", resp: []models.User{{ID: uid, Name: "stella", Age: "21"}, {ID: uid1, Name: "sam", Age: "31"}}},
+		{desc: "db error", id: "2", err: errors.DB{Err: errors.Error("db error")}},
+		{desc: "scan error", id: "3", err: errors.DB{Err: errors.Error("scan error")}},
+	}
```

Comment:
in other tests, we ain't specifying the struct fields so shouldn't we follow the same here as well?

Reply:
We should be specifying struct fields, cause that is more readable

Reply:
agreed. So we should specify them everywhere else as well for similar readability.

Reply:
Yes. This would be another issue, probably around formatting tests